test/metabase/upload_test.clj

```diff
@@ -171,42 +176,47 @@
            [" 2022-01-01T01:00:00.00Z "       (t/offset-date-time "2022-01-01T01:00+00:00") offset-dt-type]
            [" 2022-01-01t01:00:00.00Z "       (t/offset-date-time "2022-01-01T01:00+00:00") offset-dt-type]
            [" 2022-01-01 01:00:00.00Z "       (t/offset-date-time "2022-01-01T01:00+00:00") offset-dt-type]]]
-    (let [settings {:number-separators (or seps ".,")}
-          type     (#'upload/value->type string-value settings)
-          parser   (upload-parsing/upload-type->parser type settings)]
+    (let [settings   {:number-separators (or seps ".,")}
+          value-type (#'upload/value->type string-value settings)
+          ;; get the type of the column, if it were filled with only that value
+          col-type   (first (#'upload/column-types-from-rows settings 1 [[string-value]]))
```

Comment:
I made this change to the test because types of values don't correspond to a single parser anymore. Only column types correspond to a parser now. So to get a column type from a value, you need to create a fake row with a column that contains only that value, and then get the column type.